File: src/main/java/org/kite9/diagram/common/elements/RoutingInfo.java
package org.kite9.diagram.common.elements;

/**
 * Stores some kind of information about the planarization position of a vertex in order that we 
 * can route edges around it.
 * 
 * @author robmoffat
 *
 */
public interface RoutingInfo extends Comparable<RoutingInfo>{

	public String outputX();
	public String outputY();
	
	public double centerX();
	
	public double centerY();
	
	public int compareX(RoutingInfo with);
	public int compareY(RoutingInfo with);
	

}

File: visualization/kite9-visualization-common/src/visMain/kotlin/org/kite9/diagram/common/elements/RoutingInfo.kt
package org.kite9.diagram.common.elements

/**
 * Stores some kind of information about the planarization position of a vertex in order that we
 * can route edges around it.
 *
 * @author robmoffat
 */
interface RoutingInfo : Comparable<RoutingInfo> {
    fun outputX(): String
    fun outputY(): String
    fun centerX(): Double
    fun centerY(): Double
    fun compareX(with: RoutingInfo): Int
    fun compareY(with: RoutingInfo): Int
}